local/class-cli.php

```diff
@@ -168,4 +170,110 @@ private function generate_comments( $args, $assoc_args ) {
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Reprocess inbox item with ID 123
+	 *     $ wp activitypub reprocess_inbox 123
+	 *     Success: Inbox item 123 has been reprocessed.
+	 *
+	 * @param array $args The arguments.
+	 */
+	public function reprocess_inbox( $args ) {
+		$post_id = absint( $args[0] );
+
+		if ( ! $post_id ) {
+			\WP_CLI::error( 'Invalid post ID provided.' );
+		}
+
+		$post = Inbox::get( $post_id );
+
+		if ( ! $post ) {
+			\WP_CLI::error( sprintf( 'Post with ID %d not found.', $post_id ) );
+		}
+
+		\WP_CLI::log( sprintf( 'Reprocessing inbox item %d...', $post_id ) );
+
+		// Get the activity ID (GUID) and activity data.
+		$activity_id   = $post->guid;
+		$activity_data = json_decode( $post->post_content, true );
+
+		if ( ! $activity_data ) {
+			\WP_CLI::error( 'Failed to decode activity data.' );
+		}
+
+		// Only delete existing artifacts for activity types that create new ones.
+		$should_delete = isset( $activity_data['type'] ) && in_array( $activity_data['type'], array( 'Create', 'Like', 'Announce' ), true );
+
+		$deleted_comments = 0;
+		$deleted_posts    = 0;
+
+		if ( $should_delete ) {
+			// Delete comments with source_id matching the activity ID.
+			$comment = Comment::object_id_to_comment( $activity_id );
+			if ( $comment ) {
+				\wp_delete_comment( $comment->comment_ID, true );
+				++$deleted_comments;
+				\WP_CLI::log( sprintf( 'Deleted comment %d (source_id: activity ID)', $comment->comment_ID ) );
+			}
+
+			// Delete comments and posts with source_id/GUID matching the activity object ID.
+			if ( isset( $activity_data['object']['id'] ) && is_array( $activity_data['object'] ) ) {
+				$object_id = $activity_data['object']['id'];
+
+				$object_comment = Comment::object_id_to_comment( $object_id );
+				if ( $object_comment ) {
+					\wp_delete_comment( $object_comment->comment_ID, true );
+					++$deleted_comments;
+					\WP_CLI::log( sprintf( 'Deleted comment %d (source_id: object ID)', $object_comment->comment_ID ) );
+				}
+
+				// Delete ap_post with GUID matching the object ID.
+				$ap_post = Posts::get_by_guid( $object_id );
+				if ( ! \is_wp_error( $ap_post ) ) {
+					\wp_delete_post( $ap_post->ID, true );
+					++$deleted_posts;
+					\WP_CLI::log( sprintf( 'Deleted ap_post %d (GUID: object ID)', $ap_post->ID ) );
+				}
+			}
+
+			if ( $deleted_comments > 0 || $deleted_posts > 0 ) {
+				\WP_CLI::log( sprintf( 'Deleted %d comment(s) and %d post(s).', $deleted_comments, $deleted_posts ) );
+			} else {
+				\WP_CLI::log( 'No existing comments or posts found to delete.' );
+			}
+		} else {
+			\WP_CLI::log( sprintf( 'Skipping deletion for %s activity type.', $activity_data['type'] ) );
+		}
+
+		// Bypass signature verification for internal reprocessing.
+		\add_filter( 'activitypub_defer_signature_verification', '__return_true' );
+
+		// Create internal REST request to the shared inbox endpoint.
+		$request = new \WP_REST_Request( 'POST', '/' . \ACTIVITYPUB_REST_NAMESPACE . '/inbox' );
```

Comment:
I am not sure if this is always possible. Some Activities could have been sent to the Actors inbox with an empty audience. In these cases, the Activity wouldn't be reprocessed from the shared-inbox.